Fished from: http://lists.gnu.org/archive/html/qemu-devel/2014-01/msg01035.html

From: Steven Noonan <address@hidden>

This is caused by a linux/xattr.h and sys/xattr.h incompatibility:

    In file included from 
/home/snoonan/Development/qemu/include/qemu/xattr.h:27:0,
                     from fsdev/virtfs-proxy-helper.c:25:
    /usr/include/sys/xattr.h:31:3: error: expected identifier before numeric 
constant
       XATTR_CREATE = 1, /* set value, fail if attr already exists.  */
       ^
    /home/snoonan/Development/qemu/qemu/rules.mak:25: recipe for target 
'fsdev/virtfs-proxy-helper.o' failed

Moving the include around resolves it.

Signed-off-by: Steven Noonan <address@hidden>
Cc: Anthony Liguori <address@hidden>
---
 fsdev/virtfs-proxy-helper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fsdev/virtfs-proxy-helper.c b/fsdev/virtfs-proxy-helper.c
index 713a7b2..c10a085 100644
--- a/fsdev/virtfs-proxy-helper.c
+++ b/fsdev/virtfs-proxy-helper.c
@@ -12,7 +12,6 @@
 #include <sys/resource.h>
 #include <getopt.h>
 #include <syslog.h>
-#include <sys/capability.h>
 #include <sys/fsuid.h>
 #include <sys/vfs.h>
 #include <sys/ioctl.h>
@@ -23,6 +22,7 @@
 #include "qemu-common.h"
 #include "qemu/sockets.h"
 #include "qemu/xattr.h"
+#include <sys/capability.h>
 #include "virtio-9p-marshal.h"
 #include "hw/9pfs/virtio-9p-proxy.h"
 #include "fsdev/virtio-9p-marshal.h"
-- 
1.8.5.2
